lib/linebuffer.js

```diff
@@ -21,18 +20,43 @@ var bufferStats = {
 };
 var config = require('./config');
 var dccount = 0;
-var httperr;
+var httpErrorTime;
 var queue;
 
+// check response to see need for buffering into queue
+const needsRetry = (error, response, body) => {
+    if (error) {
+        debug(`req error: ${error}`);
+        if (!httpErrorTime) { httpErrorTime = Date.now(); }
+        if (bufferStats.flushcount % 10 === 0) { utils.log(`request error: ${error}`); }
+        return config.HTTP_RETRY && config.DEFAULT_HTTP_ERRORS && config.DEFAULT_HTTP_ERRORS.includes(error.code);
+    }
+
+    if (response) {
+        if (response.statusCode !== 200) {
+            debug(`req failed: ${body}`);
+            if (bufferStats.flushcount % 10 === 0) { utils.log(`request failed: ${body}`); }
+            return response.statusCode >= 500;
+        } else {
+            if (httpErrorTime) {
+                debug('req succeeded, re-enabling');
+                httpErrorTime = null;
+            }
+        }
+    }
+
+    return false;
```

Comment:
I'd prefer to return as early as possible, more readable, and less error-prone:
```
if (!response) {
    return false;
}

if (response.statusCode !== 200) {
    debug(`req failed: ${body}`);
    if (bufferStats.flushcount % 10 === 0) {
        utils.log(`request failed: ${body}`);
    }
    return response.statusCode >= 500;
}

if (httpErrorTime) {
    debug('req succeeded, re-enabling');
    httpErrorTime = null;
}
return false;
```

Reply:
This part is done but I don't know why it doesn't show as `Outdated` - can you check this part again, @weizou19?

Reply:
Done!